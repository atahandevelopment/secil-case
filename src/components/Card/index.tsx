import "./styles/card.css";
import { SingleProductTypes } from "./types";

type Props = {
  data: SingleProductTypes;
};

const Card = (props: Props) => {
  const { data } = props;
  const { title } = data;

  const truncateTitle = (text: string, maxLength: number) => {
    if (text.length > maxLength) {
      return text.substring(0, maxLength) + '...';
    } else {
      return text;
    }
  };


  return (
    <div className="card-container">
        <div className="image-container">
          <img src={`${data?.image}`} />
        </div>
        <div className="movie-name-container">
            <label>{truncateTitle(title, 40)}</label>
            <label className="price">{data?.price ? `${data?.price + ' ' + '$'}` : ''}</label>
          </div>
    </div>
  );
};

export default Card;
