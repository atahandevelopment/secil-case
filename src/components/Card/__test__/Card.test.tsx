import { expect, it } from 'vitest';
import Card from '..';
import renderer from 'react-test-renderer';

it('renders Detail component correctly', () => {
  const tree = renderer.create(<Card data={{
      id: 0,
      title: '',
      price: 0,
      description: '',
      category: '',
      image: '',
      rating: {
          rate: 0,
          count: 0
      }
  }} />).toJSON();
  expect(tree).toMatchSnapshot();
});
