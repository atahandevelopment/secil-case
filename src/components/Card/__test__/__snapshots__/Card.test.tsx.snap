// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders Detail component correctly 1`] = `
<div
  className="card-container"
>
  <div
    className="image-container"
  >
    <img
      src=""
    />
  </div>
  <div
    className="movie-name-container"
  >
    <label />
    <label
      className="price"
    />
  </div>
</div>
`;
